File: uikit/src/main/java/io/agora/chat/uikit/chat/interfaces/IChatExtendMenu.java
package io.agora.chat.uikit.chat.interfaces;

public interface IChatExtendMenu {

    void clear();

    /**
     * Set the order of item
     * @param itemId
     * @param order
     */
    void setMenuOrder(int itemId, int order);

    /**
     * Add new extended functions
     * @param name
     * @param drawableRes
     * @param itemId
     */
    void registerMenuItem(String name, int drawableRes, int itemId);

    /**
     * Add new extended functions
     * @param name
     * @param drawableRes
     * @param itemId
     * @param order
     */
    void registerMenuItem(String name, int drawableRes, int itemId, int order);

    /**
     * Add new extended functions
     * @param nameRes
     * @param drawableRes
     * @param itemId
     */
    void registerMenuItem(int nameRes, int drawableRes, int itemId);

    /**
     * Add new extended functions
     * @param nameRes
     * @param drawableRes
     * @param itemId
     * @param order
     */
    void registerMenuItem(int nameRes, int drawableRes, int itemId, int order);

    /**
     * Set item click listener
     * @param listener
     */
    void setEaseChatExtendMenuItemClickListener(EaseChatExtendMenuItemClickListener listener);
}

File: ease-im-kit/src/main/kotlin/com/hyphenate/easeui/feature/chat/interfaces/IChatExtendMenu.kt
package com.hyphenate.easeui.feature.chat.interfaces

interface IChatExtendMenu {
    fun clear()

    /**
     * Set the order of item
     * @param itemId
     * @param order
     */
    fun setMenuOrder(itemId: Int, order: Int)

    /**
     * Add new extended functions
     * @param name
     * @param drawableRes
     * @param itemId
     * @param order
     */
    fun registerMenuItem(name: String?, drawableRes: Int, itemId: Int, order: Int = 0,titleColor:Int = -1,resourceTintColor:Int = -1)

    /**
     * Add new extended functions
     * @param nameRes
     * @param drawableRes
     * @param itemId
     * @param order
     */
    fun registerMenuItem(nameRes: Int, drawableRes: Int, itemId: Int, order: Int = 0,titleColor:Int = -1,resourceTintColor:Int = -1)

    /**
     * Set item click listener
     * @param listener
     */
    fun setEaseChatExtendMenuItemClickListener(listener: ChatUIKitExtendMenuItemClickListener?)
}